### PR TITLE
docs: further changes to the glossary

### PR DESCRIPTION
**Pull request by tomwaterton**
### Further changes to the Glossary page:
- In the Glossary page description, added a link to our capitalization guidance
- Changed all links to other glossary entries to be shown in lowercase (to match how the entries are themselves listed)
- Ensured all links to other glossary terms are written in the format: "Compare (term)"
- Ensured that all links to guidance from the Patterns > Common actions page are written in the format: "See also the (action) pattern"
- Where an entry lists more than one related glossary term, listed those related terms in alphabetical order
- Tweaked the wording of some of the glossary entries
- Used the monospace code format for "do not use" terms
- Changed the styling (padding/margin) in the `glossary-lists.scss` file so that the padding comes from the main term entry ("word-heading") rather than from either the term description ("desc") or guidance ("subtext") paragraphs because not all entries have both "desc" and "subtext" text.

### CODE REVIEW

File: src/data/guidelines/glossary.js [hunk 1]
```diff
@@ -40,31 +39,31 @@ module.exports = {
       cancel: {
         desc: 'Stops the current action and closes the dialog.',
         subtext:
-          'Warn the user of any possible negative consequences of stopping an action from progressing, such as data corruption. See [Reset](#reset) and the [Cancel action](/patterns/common-actions#cancel).',
+          'Warn the user of any possible negative consequences of stopping an action from progressing, such as data corruption. Compare [reset](#reset). See also the [Cancel pattern](/patterns/common-actions#cancel).',
       },
       clear: {
         desc:
-          'This action clears all the fields or selections. Also deletes the contents of a document, such as a log. Typically the default selection or value is re-established for controls that always have a selection or value, such as Radio Buttons.',
+          'This action clears all the fields or selections. Also deletes the contents of a document, such as a log. Typically the default selection or value is re-established for controls that always have a selection or value, such as radio buttons.',
```
Comment (connor-leech) [inline src/data/guidelines/glossary.js]:
```suggestion
          'This action clears all the fields or selections. Also deletes the contents of a document, such as a log. Typically the default selection or value is re-established for controls that always have a selection or value, such as [radio buttons](/components/radio-button/usage).',
```

File: src/data/guidelines/glossary.js [hunk 2]
```diff
@@ -257,51 +259,51 @@ module.exports = {
       redo: {
         desc: 'Redoes an undo action.',
         subtext:
-          'Likely used only as a Tooltip on an icon button. See [Undo](#undo).',
+          'Likely used only as a tooltip on an icon button. Compare [undo](#undo).',
```
Comment (connor-leech) [inline src/data/guidelines/glossary.js]:
```suggestion
          'Likely used only as a [tooltip](/components/tooltip/usage) on an icon button. Compare [undo](#undo).',
```

**Comment by connor-leech** [comment]
This looks great. I added a few links to components referenced, but besides those, I think it's good to go

File: src/components/GlossaryComponent/GlossaryList/glossary-list.scss [hunk 3]
```diff
@@ -2,17 +2,23 @@
...
   text-transform: lowercase;
 }
 
+.glossary-entry__word-heading {
+  padding-top: 0.75rem;
+}
+
+.glossary-entry__desc {
+  padding: 0px;
+  margin-top: 0.25rem;
```
Comment (alisonjoseph) [inline src/components/GlossaryComponent/GlossaryList/glossary-list.scss]:
```suggestion
  margin-top: $spacing-04;
```

File: src/components/GlossaryComponent/GlossaryList/glossary-list.scss [hunk 4]
```diff
@@ -2,17 +2,23 @@
...
-.glossary-entry__main-heading {
-  margin-bottom: $spacing-06;
-}
-
 .glossary-entry__main-heading span {
   text-transform: lowercase;
 }
 
+.glossary-entry__word-heading {
+  padding-top: 0.75rem;
```
Comment (alisonjoseph) [inline src/components/GlossaryComponent/GlossaryList/glossary-list.scss]:
```suggestion
  padding-top:$spacing-04;
```

File: src/components/GlossaryComponent/GlossaryList/glossary-list.scss [hunk 5]
```diff
@@ -2,17 +2,23 @@
...
+  margin-top: 0.25rem;
+}
+
 .glossary-entry__subtext {
   @include carbon--type-style('body-short-02');
   color: $text-02;
   font-style: italic;
-  margin-top: $spacing-02;
+  padding: 0px;
+  margin-top: 0.25rem;
```
Comment (alisonjoseph) [inline src/components/GlossaryComponent/GlossaryList/glossary-list.scss]:
```suggestion
  margin-top: $spacing-02;
```

**Comment by alisonjoseph** [comment]
Couple comments where spacing tokens could be used instead of hard coded values. I also tagged design to review/approve this since we have some spacing changes. thanks!

**Pull request merged**